### ISSUE TITLE
Questions regarding experiments settings and explanations in the paper

### ISSUE THREAD

**Issue by b10815061** [hengRUC/VSP#2]
Hello,

Congrats on the great work! And thanks for opening source.

When I read the paper, I found something quite obscure to me, could you please help me with that?

1. In the 3.3 section. You mentioned "Given N triplets sampled from N videos, through encoder ...", so for one video will there be one sample or multiple sample? (Cause when I check the code, the config constructs 5 bridge for the pouring dataset., which is 5 triplets for 1 video.)

2. In the **Implementation Details** in section 4.1. The training setup of batch size is 128, which is also not consistent with the config the code provided. 

I'm not sure if I understood these parts correctly. Thank you!

**Comment by hengRUC** [comment]
Hi,
Thank you for your kind words and your interest in the work. I'd be happy to clarify the points you mentioned.

1. Based on experimental experience, using multiple Brownian bridges in a distributed environment is beneficial for training. Our setting in the paper is to better describe Process based Comparative Loss. In fact, if your computing resources do not allow distributed training, a single Brownian bridge is also acceptable.
2. Regarding the batch size in the implementation details, which is a hyperparameter that can be defined by oneself before training. We used 128 in our experiments.

If you have further questions or need more clarification, feel free to ask. I'm here to help!

**Comment by b10815061** [comment]
Well, but how do we assure that the brownian bridges are overlapped just like the paper described in ablation study if we have only 1 brownian bridge for a video?
I thought the bridges should be overlapping with each other with a 20% ratio just like Figure2 (b) illustrated. 

Am I missing something here?

**Comment by hengRUC** [comment]
Your understanding is correct. To achieve the desired overlap rate, it is necessary to segment the video with a 20% overlap between segments, a relatively simple video processing pretreatment.

**Comment by b10815061** [comment]
So in conclusion, several bridges could be generated by one video. 
Say we have a batch size of 2, in side this, 5 (a,b,c,d,e) bridges were generated by Video A and 8 bridges were generated by Video B. 
In **equation 3**. If we are using VSP and VSP-P, the positive sample will comes from the ith bridge from A. (For example a) And all the other negative samples will all come from other bridges from A as well as those bridges from B. (So positive sample count is 1 and negative samples count is 12). 

Am I correct? 
Thank you for your patience.

**Comment by hengRUC** [comment]
Basically correct, it needs to be added that the video segmentation in VSP is customized, while in VSP-P it is divided according to annotations. Additionally, the number of Brown bridges in each video in the batch should be the same. Because the data is based on videos, it is necessary to ensure the alignment of data units in **dataloader**. If video- pre-processing is used, and the data is in segments, there is a small probability that your example will appear, but the batch size should be (5+8=13) instead of 2.

I hope the above answers are helpful.

**Comment by b10815061** [comment]
Ah I see! So this explain why the batch size was set to 128 in the paper, originally I thought the batch size was refer to the number of video just as CARL and TCC did and was kinda confused how to batch this much of video into the memory. 

Thanks for the explanation, it was very clear!
